src/with-theme.js

```diff
@@ -0,0 +1,38 @@
+import React, {Component} from 'react'
+
+import {CHANNEL} from './theme-provider'
+import {PropTypes} from './react-compat'
+
+export default function withTheme(ComponentToTheme) {
+  class ThemedComponent extends Component {
+    state = {theme: {}}
+    setTheme = theme => this.setState({theme})
+
+    componentWillMount() {
+      if (this.context[CHANNEL]) {
+        this.setState({theme: this.context[CHANNEL].getState()})
+      }
+    }
+
+    componentDidMount() {
+      if (this.context[CHANNEL]) {
```

Comment:
I added this back in under a process.env if statement so it should be tree shakeable once the build part is figured back out. I had to ignore the console statement for eslint and the if statement for instanbul since there is no way to test that statement safely (that I am aware of).